src/classes/GhosteryDebug.js

```diff
@@ -0,0 +1,876 @@
+/**
+ * Ghostery Debug Class
+ *
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2020 Ghostery, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import abtest from './ABTest';
+import account from './Account';
+import cmp from './CMP';
+import confData from './ConfData';
+import globals from './Globals';
+import tabInfo from './TabInfo';
+import foundBugs from './FoundBugs';
+import PromoModals from './PromoModals';
+import { alwaysLog, isLog, activateLog } from '../utils/common';
+import { capitalize, getObjectSlice, pickRandomArrEl } from '../utils/utils';
+
+/**
+ * @class for debugging Ghostery via the background.js console.
+ * @memberof BackgroundClasses
+ */
+
+const OBJECT_OUTPUT_STYLE = true;
+const STRING_OUTPUT_STYLE = false;
+
+const THANKS = 'Thanks for using Ghostery';
+const UP_REMINDER = 'Remember you can press up to avoid having to retype your previous command';
+const CSS_SUBHEADER = 'css_subheader__';
+const CSS_MAINHEADER = 'css_mainheader__';
+const CSS_HIGHLIGHT = 'css_highlight__';
+const OUTPUT_COLUMN_WIDTH = 40;
+
+/**
+ * Class that implements an interactive console debugger.
+ *
+ * @since 8.5.3
+ *
+ * @memberOf  BackgroundClasses
+ */
+class GhosteryDebug {
+	// ToC
+	// Search for these strings to quickly jump to their sections
+	// [[Output styling, formatting, and printing]]
+	// [[Help CLI & strings]]
+	// [[Main Actions]]
+	// [[Settings Actions]]
+
+	constructor() {
+		// Public settings methods are defined in a public instance field in the [[Settings Actions]] section
+		// These are the private settings methods and properties
```

Comment:
Should these be class instance variables rather than having them set on `this.settings`? I'm able to access `ghostery.settings._toggleSettingHelper()` from the console.

Reply:
Moved `_toggleSettingsHelper()` to a class instance variable. It doesn't appear to be visible from the console anymore.